core/src/iam/jwks.rs

```diff
@@ -108,12 +109,55 @@ pub(super) async fn config(
 		}
 	};
 
-	// Check if algorithm specified is supported
+	// Use algorithm provided, if specified
+	// This parameter is not required to be present, although is usually expected
+	// When missing, tokens must be validated using only the required key type parameter
+	// This is discouraged, as it requires relying on the algorithm specified in the token
+	// Source: https://datatracker.ietf.org/doc/html/rfc7517#section-4.4
 	let alg = match jwk.common.algorithm {
 		Some(alg) => alg,
+		// If not specified, use the algorithm provided in the token header
+		// It is critical that the JWT library prevents the "none" algorithm from being used
+		// Reference: https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/#Meet-the--None--Algorithm
+		// In the case of "jsonwebtoken", the "none" algorithm is not part of the enumeration
+		// Source: https://docs.rs/jsonwebtoken/latest/jsonwebtoken/enum.Algorithm.html
+		// Confirmation: https://github.com/Keats/jsonwebtoken/issues/381
 		_ => {
-			warn!("Invalid value for parameter 'alg' in JWK object: '{:?}'", jwk.common.algorithm);
-			return Err(Error::InvalidAuth); // Return opaque error
+			// Ensure that the algorithm specified in the token matches the key type defined in the JWK
+			match (&jwk.algorithm, token_alg) {
+				(RSA(_), RS256 | RS384 | RS512 | PS256 | PS384 | PS512) => token_alg,
```

Comment:
should we fail if the key_type is none of the ones in the list?

Reply:
Unless I misunderstood how this works in Rust, this should fail in compile type if there is an enumerator variant that is not included in the enumerator, since I have not included a catch-all arm in the match statement. As it is, the match statement is currently comprehensive of all the variants implemented in the library, so that if a new variant is introduced in a future update compilation will fail and we will be forced to look into the cause.